### PR TITLE
new implementation of style attribute css parsing

### PR DESCRIPTION
**Pull request by neraliu**
this implementation includes the following
1) using the jison to create the parser for parsing the style attribute
2) coding a html decoding logic for delimiter that will affect CSS parsing
3) all related unit test cases.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: dist/secure-handlebars.js [hunk 1]
```diff
@@ -9792,7 +9840,989 @@ module.exports = ContextParserHandlebars;
...
+/* this == yyval */
+
+var $0 = $$.length - 1;
+switch (yystate) {
+case 1:
+
+      this.$ = [];
+      var r = this.$;
+      $$[$0-1] !== null? this.$.push($$[$0-1]) : '';
+      $$[$0] !== null? $$[$0].forEach(function(e) { r.push(e); }) : ''
```
Comment (adon-at-work) [inline dist/secure-handlebars.js]: IE compatibility issue

File: tests/samples/files/handlebarsjs_filter_attr_value_style_002.hbs [hunk 2]
```diff
@@ -6,3 +6,6 @@
 <div style ="   {{style4}}  ">
 <div style ='   {{style5}}  '>
 <div style =    {{style6}}  >
+<div style ="&#0*32;&#x20;&#9;&#x09;&Tab;&#10;&#x0a;&NewLine;&#12;&#x0c;&#13;&#x0d;{{style7}}&#0*32;  ">
+<div style ='&#0*32;&#x20;&#9;&#x09;&Tab;&#10;&#x0a;&NewLine;&#12;&#x0c;&#13;&#x0d;{{style8}}&#0*32;  '>
+<div style =&#0*32;&#x20;&#9;&#x09;&Tab;&#10;&#x0a;&NewLine;&#12;&#x0c;&#13;&#x0d;{{style9}}&#0*32;    >
```
Comment (adon-at-work) [inline tests/samples/files/handlebarsjs_filter_attr_value_style_002.hbs]: are those `0*` intended in `&#0*32;` for test cases?

Comment (neraliu) [inline tests/samples/files/handlebarsjs_filter_attr_value_style_002.hbs]: yes the zero is added intentionally. however, those cases are not needed anymore, as i have another set of unit test to cover.

but we need to add back one test case to catch the parse error when there is ';' left after the html decoding.

File: src/css-utils.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+* no matter decode or not if it does not belong to the STRING token.
+*
+* These function decode more than we need to prevent CSS parsing error.
+*
+* Reference:
+* http://www.w3.org/TR/css-style-attr/
+*
+*/
+CSSParserUtils.reCharReferenceDecode = /&([a-z]{2,};?)|&#0*(x[a-f0-9]+|[0-9]+);?/ig;
+CSSParserUtils.htmlStyleAttributeValueEntitiesDecode = function(str) {
```
Comment (neraliu) [inline src/css-utils.js]: TODO: using client browser for html decode will be added in next PR.

File: dist/secure-handlebars.js (hunk 1 above)
Comment (neraliu) [inline dist/secure-handlebars.js]: add the forEach in the polyfill.js, but it is not recommended to use secure-handlebars to compile the template in the client side.

**Comment by neraliu** [comment]
i am fixing the build error

**Comment by neraliu** [comment]
closing this PR and use new develop branch in master repo

**Pull request closed without merging**